conjureplugin/config/internal/v2/config.go

```diff
@@ -0,0 +1,121 @@
+	GroupID string `yaml:"group-id,omitempty"`
+	// AllowConflictingOutputDirs downgrades output directory conflicts from errors to warnings.
+	// Defaults to false (conflicts are errors).
+	AllowConflictingOutputDirs bool                           `yaml:"allow-conflicting-output-dirs,omitempty"`
+	ProjectConfigs             map[string]SingleConjureConfig `yaml:"projects"`
+}
+
+type SingleConjureConfig struct {
+	// OutputDir specifies the base output directory for generated code.
+	// Defaults to "internal/generated/conjure" if not specified.
+	// By default, code is generated into {OutputDir}/{ProjectName}/ unless OmitTopLevelProjectDir is true.
+	OutputDir string          `yaml:"output-dir,omitempty"`
+	IRLocator IRLocatorConfig `yaml:"ir-locator"`
+	// GroupID is the group ID for this project. If not specified, the top-level group-id is used.
+	GroupID string `yaml:"group-id,omitempty"`
+	// Publish specifies whether or not the IR specified by this project should be included in the publish operation.
+	// If this value is not explicitly specified in configuration, it is treated as "true" for YAML sources of IR and
+	// "false" for all other sources.
+	Publish *bool `yaml:"publish,omitempty"`
+	// Server indicates if we will generate server code. Currently this is behind a feature flag and is subject to change.
+	Server bool `yaml:"server,omitempty"`
+	// CLI indicates if we will generate cobra CLI bindings. Currently this is behind a feature flag and is subject to change.
+	CLI bool `yaml:"cli,omitempty"`
+	// AcceptFuncs indicates if we will generate lambda based visitor code.
+	// Currently this is behind a feature flag and is subject to change.
+	AcceptFuncs *bool `yaml:"accept-funcs,omitempty"`
+	// Extensions contain metadata for consumption by assets of type `conjure-ir-extensions-provider`.
+	Extensions map[string]any `yaml:"extensions,omitempty"`
+	// OmitTopLevelProjectDir skips creating the {ProjectName} subdirectory.
+	// When false (default), generates into {OutputDir}/{ProjectName}/.
+	// When true, generates directly into {OutputDir}/.
+	OmitTopLevelProjectDir bool `yaml:"omit-top-level-project-dir,omitempty"`
+	// SkipDeleteGeneratedFiles skips cleanup of old generated files before regeneration.
+	// When false (default), deletes all Conjure-generated files in the output directory tree before regenerating.
+	// When true, preserves v1 behavior (no cleanup).
+	SkipDeleteGeneratedFiles bool `yaml:"skip-delete-generated-files,omitempty"`
+}
+
+// ResolvedOutputDir returns the final output directory path where generated code will be written.
+// It applies the following logic:
+// 1. Uses OutputDir if specified, otherwise defaults to DefaultOutputDir ("internal/generated/conjure")
+// 2. Appends the projectName subdirectory unless OmitTopLevelProjectDir is true
+// 3. Normalizes the path with filepath.Clean
+func (s SingleConjureConfig) ResolvedOutputDir(projectName string) string {
```

Comment:
Recommend to declare methods on pointer receivers unless there's a specific reason not to do so

```suggestion
func (s *SingleConjureConfig) ResolvedOutputDir(projectName string) string {
```